src/dbus_api/api.rs

```diff
@@ -371,33 +333,8 @@ fn rename_filesystem(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
             return_message.append3(default_return, rc, rs)
         }
         Ok(RenameAction::Renamed) => {
-            let return_value = MessageItem::Bool(true);
-            let removed = dbus_context.filesystems
-                .borrow_mut()
-                .remove_by_second(&(pool_name.clone(), filesystem_name.into()));
-            match removed {
-                Some((removed_object_path, _)) => {
-                    if object_path.to_string() == removed_object_path {
-                        dbus_context.filesystems
-                            .borrow_mut()
-                            .insert(removed_object_path, (pool_name.into(), new_name.into()));
-                        let (rc, rs) = ok_message_items();
-                        return_message.append3(return_value, rc, rs)
-                    } else {
-                        let error_message = format!("wrong dbus object_path for renamed \
-                                                    filesystem");
-                        let (rc, rs) = code_to_message_items(DbusErrorEnum::INTERNAL_ERROR,
-                                                             error_message);
-                        return_message.append3(return_value, rc, rs)
-                    }
-                }
-                None => {
-                    let error_message = format!("no dbus object path for renamed filesystem");
-                    let (rc, rs) = code_to_message_items(DbusErrorEnum::INTERNAL_ERROR,
-                                                         error_message);
-                    return_message.append3(return_value, rc, rs)
-                }
-            }
+            let (rc, rs) = ok_message_items();
```

Comment:
Yes, the new, simplified one-directional tables are a huge win when renaming pools and filesystems.